pkg/kubelet/kubelet_getters_test.go

```diff
@@ -91,3 +96,111 @@ func TestKubeletDirs(t *testing.T) {
+	kubelet, expectedPods, expectedRunningPods := newKubeletAndPods(testKubelet)
+
+	testCases := []struct {
+		name       string
+		assertFunc func(t *testing.T)
+	}{
+		{
+			name: "GetPods",
+			assertFunc: func(t *testing.T) {
+				got := kubelet.GetPods()
+				assert.Equal(t, len(expectedPods), len(got))
+			},
+		},
+		{
+			name: "GetRunningPods",
+			assertFunc: func(t *testing.T) {
+				got, err := kubelet.GetRunningPods()
+				if err != nil {
+					t.Fatalf("unexpected error: %v", err)
+				}
+				assert.Equal(t, len(expectedRunningPods), len(got))
+			},
+		},
+		{
+			name: "GetPodByFullName",
+			assertFunc: func(t *testing.T) {
+				got, ok := kubelet.GetPodByFullName("pod0_")
+				assert.Equal(t, true, ok)
+				assert.Equal(t, expectedPods[0], got)
+			},
+		},
+		{
+			name: "GetPodByName",
+			assertFunc: func(t *testing.T) {
+				got, ok := kubelet.GetPodByName("", "pod1")
+				assert.Equal(t, true, ok)
+				assert.Equal(t, expectedPods[1], got)
+			},
+		},
+		{
+			name: "GetPodByCgroupfs",
+			assertFunc: func(t *testing.T) {
+				got, ok := kubelet.GetPodByCgroupfs("10001")
+				assert.Equal(t, false, ok)
+				assert.Equal(t, (*v1.Pod)(nil), got)
+			},
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			tc.assertFunc(t)
+		})
+	}
+}
+
+func TestGetHost(t *testing.T) {
+	testKubelet := newTestKubelet(t, false /* controllerAttachDetachEnabled */)
+	defer testKubelet.Cleanup()
+	kubelet := testKubelet.kubelet
+
+	got := kubelet.GetHostname()
+	exp := "127.0.0.1"
+	assert.Equal(t, exp, got, "GetHostname")
+
+	gotIPs, err := kubelet.GetHostIPs()
+	assert.NoError(t, err, "GetHostIPs error")
+	expIPs := []net.IP{
+		net.ParseIP("127.0.0.1"),
+	}
+	assert.Equal(t, expIPs, gotIPs, "GetHostIPs")
+
+	kubelet.nodeName = "127.0.0.2"
+	_, err = kubelet.GetHostIPs()
+	expErr := fmt.Errorf("Node with name: 127.0.0.2 does not exist")
+	assert.ErrorAs(t, err, &expErr, "GetHostIPs")
+}
+
+func newKubeletAndPods(testKubelet *TestKubelet) (*Kubelet, []*v1.Pod, []*containertest.FakePod) {
+	kubelet := testKubelet.kubelet
+
+	expectedPods := newTestPods(2)
+	expectedPods[0].ObjectMeta.Annotations = make(map[string]string)
+	expectedPods[0].ObjectMeta.Annotations["kubernetes.io/config.source"] = "file"
+
+	kubelet.podManager.SetPods(expectedPods)
+	kubelet.statusManager.SetPodStatus(expectedPods[0], v1.PodStatus{Phase: v1.PodSucceeded})
+
+	return kubelet, expectedPods, testKubelet.fakeRuntime.PodList
+}
+
+func TestGetCadvisor(t *testing.T) {
+	testKubelet := newTestKubelet(t, false /* controllerAttachDetachEnabled */)
+	defer testKubelet.Cleanup()
+	kubelet := testKubelet.kubelet
+
+	got, err := kubelet.GetVersionInfo()
+	assert.NoError(t, err, "GetVersionInfo error")
+	exp := &cadvisorapi.VersionInfo{
+		KernelVersion:      "3.16.0-0.bpo.4-amd64",
+		ContainerOsVersion: "Debian GNU/Linux 7 (wheezy)",
+		DockerVersion:      "1.13.1",
+	}
+	assert.Equal(t, exp, got, "GetVersionInfo")
+}
```

Comment:
Is `GetRequestedContainersInfo` testable? It's also uncovered.

Reply:
Done. Improve coverage, but it doesn’t make sense for the time being.
There are many examples of fake returning `zero values`, which require additional optimization, but not here. :-)